src/sync.js

```diff
@@ -237,8 +237,8 @@ class Sync {
       if (err) { log.error(err) }
       if (err && err.code === 'ENOSPC') {
         callback(new Error('The disk space on your computer is full!'))
-      } else if (err && err.message === REVOKED) {
-        callback(err)
+      } else if (err && err.status === 400) {
+        callback(new Error(REVOKED))
```

Comment:
Yes, it will be detected as a revoked device. The user can then unlink the device and redo the OAuth 2 flow. It's far from ideal, but not a blocker too.